test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -105,28 +109,114 @@ def test_revert_pbsconf_remotemom(self):
         self.assertEqual(pbs_conf_val["PBS_START_COMM"], "1")
 
         # Check that the remote mom's pbs.conf has mom start bit on
-        mom = self.moms.values[0]
-        pbs_conf_filepath = self.du.get_pbs_conf_file(mom.hostname)
-        pbs_conf_val = self.du.parse_pbs_config(mom.hostname,
+        pbs_conf_filepath = self.du.get_pbs_conf_file(remotemom.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname,
                                                 pbs_conf_filepath)
         self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
 
         # Now set it to 0 and restart the mom
-        self.mom.pi.stop(mom.hostname)
+        remotemom.pi.stop(remotemom.hostname)
         pbs_conf_val["PBS_START_MOM"] = "0"
         self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
-        mom.pi.start(mom.hostname)
+        remotemom.pi.start(remotemom.hostname)
 
         # Confirm that the mom is down
-        self.assertFalse(mom.isUp())
+        self.assertFalse(remotemom.isUp())
 
         # Now call revert_pbsconf() & restart mom again
-        self.mom.pi.stop(mom.hostname)
         self.revert_pbsconf()
-        mom.pi.restart(mom.hostname)
-        mom.pi.start(mom.hostname)
 
         # Confirm that the mom came up and start bit is 1
-        self.assertTrue(mom.isUp())
-        pbs_conf_val = self.du.parse_pbs_config(mom.hostname)
+        self.assertTrue(remotemom.isUp())
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname)
         self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+
+    def test_revert_pbsconf_corelimit(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf() when
+        PBS_CORE_LIMIT is reverted
+        """
+        pbs_conf_filepath = self.du.get_pbs_conf_file(self.server.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname,
+                                                pbs_conf_filepath)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the setUp already ran, check that PBS_CORE_LIMIT is set to
+        # unlimited
+        self.assertEqual(pbs_conf_val["PBS_CORE_LIMIT"], "unlimited")
+
+        # Now, set the core limit to 0 and restart PBS
+        self.server.pi.stop()
+        pbs_conf_val["PBS_CORE_LIMIT"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+        self.server.pi.start()
+
+        # First, check that there's no existing core file in mom_priv
+        parent_dir = os.getcwd()
+        snapdirs = []
+        with PBSSnapUtils(out_dir=parent_dir, acct_logs=0, daemon_logs=0,
```

Comment:
Why do we need to take snapshots? Can we not run an 'ls' command on the mom_priv directory?

Reply:
we can, but how will we know which file is a core file?

Reply:
I thought they all start with prefix core and if there are already existing core files in the 'priv' directory then daemons store the previous core files by appending an integer value to their name, like core.12434 will become core_001

Reply:
nope, I've seen core file names start with 'vgcore' or something completely different, it depends on how the system is configured: https://sigquit.wordpress.com/2009/03/13/the-core-pattern/

Reply:
hmm .. I can see what you are trying to say, why not just go for the simple approach as it might be good enough .. since this is just a PTL test which will most probably run in a controlled environment anyways, I'll change it to the way you are suggesting